menu/cbs/menu_cbs_ok.c

```diff
@@ -6788,6 +6799,50 @@ static int action_ok_push_dropdown_item_input_description_kbd(
    return action_cancel_pop_default(NULL, NULL, 0, 0);
 }
 
+static int action_ok_push_dropdown_item_netplay_mitm_server(const char *path,
+      const char *label, unsigned type, size_t idx, size_t entry_idx)
+{
+#ifdef HAVE_NETWORKING
+
+   const char *menu_path        = NULL;
+   enum msg_hash_enums enum_idx;
+   rarch_setting_t     *setting;
+
+   unsigned list_len = ARRAY_SIZE(netplay_mitm_server_list);
+   bool found        = false;
+   int offset        = 0;
+   unsigned i;
+
+   menu_entries_get_last_stack(&menu_path, NULL, NULL, NULL, NULL);
+   enum_idx = (enum msg_hash_enums)atoi(menu_path);
+   setting  = menu_setting_find_enum(enum_idx);
+
+   if (!setting)
+      return menu_cbs_exit();
+
+   for (i = 0; i < list_len; i++)
+   {
+      /* find the currently selected server in the list */
+      if (string_is_equal(path, netplay_mitm_server_list[i].description))
+      {
+         found  = true;
+         offset = i;
+         break;
+      }
+   }
```

Comment:
Get rid of this loop.

Reply:
Debug shows that `idx` and `entry_idx` arguments of `action_ok_push_dropdown_item_netplay_mitm_server` contain equal values. Which one is better to use to get index of the currently selected item in the drop down list?

Reply:
Ok, i think it's `idx`

Reply:
Neither. I've updated that suggestion.

But the correct one would be entry_idx, idx should always be 0 in this case.

Reply:
I must use entry_idx then. As `path` contains Description of a MITM server, like "New Yourk, USA" while setting contain its name, like "nyc".

Reply:
use label, as we now pass the name member as the label (if you've checked my suggestion on menu_displaylist.c).